larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -209,43 +144,68 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
       conv_params->filter_height, conv_params->filter_width,
       conv_params->padding_type, &out_height, &out_width);
 
-  CalculateActivationRange(conv_params->fused_activation_function,
-                           &conv_params->output_activation_min,
-                           &conv_params->output_activation_max);
+  if (output->type == kTfLiteInt32) {
+    TF_LITE_ENSURE_EQ(context, NumDimensions(thresholds), 1);
+    TF_LITE_ENSURE_EQ(context, thresholds->type, kTfLiteInt32);
+    TF_LITE_ENSURE_EQ(context, SizeOfDimension(thresholds, 0),
+                      conv_params->channels_out);
+    TF_LITE_ENSURE_MSG(context,
+                       conv_params->padding_type != kTfLitePaddingSame ||
+                           conv_params->pad_value == 1,
+                       "Writing bitpacked output is only supported with "
+                       "valid or one-padding.");
+  } else {
+    TF_LITE_ENSURE_EQ(context, NumDimensions(post_activation_multiplier), 1);
+    TF_LITE_ENSURE_EQ(context, NumDimensions(post_activation_bias), 1);
+    TF_LITE_ENSURE_EQ(context, SizeOfDimension(post_activation_multiplier, 0),
+                      conv_params->channels_out);
+    TF_LITE_ENSURE_EQ(context, SizeOfDimension(post_activation_bias, 0),
+                      conv_params->channels_out);
+  }
 
   if (output->type == kTfLiteInt8) {
+    // With 8-bit output, we allow int8 or float32 `post_activation` values
+    TF_LITE_ENSURE(context,
+                   post_activation_multiplier->type == kTfLiteFloat32 ||
+                       post_activation_multiplier->type == kTfLiteInt8);
+    TF_LITE_ENSURE(context, post_activation_bias->type == kTfLiteFloat32 ||
+                                post_activation_bias->type == kTfLiteInt8);
```

Comment:
I think we now also specified that these are always `float32` and not `int8`, so this can also be slightly simplified.
(I'm not sure if this'll pass linting)
```suggestion
    // With 8-bit output we require float32 `post_activation` values
    TF_LITE_ENSURE_EQ(context,
                   post_activation_multiplier->type, kTfLiteFloat32);
    TF_LITE_ENSURE_EQ(context, post_activation_bias->type, kTfLiteFloat32);
```

Reply:
Ah cool, I hadn't realised that, thanks :+1: Will update.